source/javascripts/index.js

```diff
@@ -71,24 +75,63 @@ for (let index = 0; index < carouselList.length; index++) {
 
 // initiate custom calendar carousels
 const carouselCalendarList = document.querySelectorAll(
-  '[data-calendar-splide]'
+  "[data-calendar-splide]"
 );
 for (let index = 0; index < carouselCalendarList.length; index++) {
   const carousel = carouselCalendarList[index];
   const carouselInstances = [];
   carouselInstances[index] = new CarouselCalendar(carousel);
 }
 
-const chartWrap = document.getElementsByClassName('chartWrap');
+// setup yt videos
+const YT_SERVICE = "youtube.com";
+const loadYouTubeVideo = (videoId) => {
+  console.log("videoId", videoId);
+  const videoEl = document.getElementById(videoId);
+  const url = videoEl.dataset.url;
+  console.log("videoUrl", url);
+  const video = VideoPlayer.getOrCreateInstance(videoEl);
+  console.log("video", video);
+  video.setYouTubeVideo(url);
+};
+const manageVideo = (overlay) => {
+  const videoLoadButton = overlay.querySelector("button");
+  const videoId = videoLoadButton.dataset.videoid;
+  if (cookies.isChoiceRemembered(YT_SERVICE)) {
+    loadYouTubeVideo(videoId);
+    const overlayAccept = new AcceptOverlay(overlay, {
+      service: YT_SERVICE,
+    });
+    overlayAccept.hide();
+  } else {
+    videoLoadButton.addEventListener("click", () => {
+      const checkbox = overlay.querySelector("input");
+      if (checkbox.checked) {
+        cookies.rememberChoice(YT_SERVICE, true);
+      }
+      loadYouTubeVideo(videoId);
+      const overlayAccept = new AcceptOverlay(overlay, {
+        service: YT_SERVICE,
+      });
+      overlayAccept.hide();
+    });
+  }
+};
+
+const overlay = document.querySelector("[data-bs-accept-overlay]");
+if (overlay) {
+  manageVideo(overlay);
+}
+
+//setup charts
+const chartWrap = document.getElementsByClassName("chartWrap");
```

Comment:
These chart changes should be in a different PR. They also look like purely formatting changes.